orderer/consensus/smartbft/heartbeatmonitor.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package smartbft
+
+import (
+	"sync"
+	"time"
+
+	"github.com/hyperledger/fabric/common/flogging"
+)
+
+// Role indicates if this node sends or receives heartbeats
+type Role bool
+
+// A node could either be a sender or a receiver
+const (
+	HeartbeatSender   Role = false
+	HeartbeatReceiver Role = true
+)
+
+type HeartbeatMonitor struct {
+	rpc               RPC
+	scheduler         <-chan time.Time
+	inc               chan uint64
+	role              Role
+	senders           []uint64
+	receivers         []uint64
+	lastReceivedTimes []time.Time
+	hbTimeout         time.Duration
+	hbCount           uint64
+	lastTick          time.Time
+	lastHeartbeat     time.Time
+	logger            *flogging.FabricLogger
```

Comment:
Better to just define the logger interface and then you can use the zap based sugared logger like we do in the tests of the consensus library. 

Then you will also see the prints.